<launch>
  <!-- Name of the node -->
  <arg name="eskf_node" default="attitude_eskf"/>

  <!-- IMU topic to use -->
  <arg name="imu_topic" default="/imu/imu"/>

  <!-- Magnetic field topic to use -->
  <arg name="field_topic" default="/imu/magnetic_field"/>

  <!-- Enable magnetometer -->
  <arg name="enable_magnetometer" default="false"/>

  <!-- Parameters for ESKF -->
  <arg name="param_file" default="$(find kr_attitude_eskf)/config/params.yaml"/>

  <!-- Broadcast the TF -->
  <arg name="broadcast_frame" default="true"/>

  <node pkg="kr_attitude_eskf" name="$(arg eskf_node)" type="kr_attitude_eskf" output="screen">
    <param name="imu_topic" type="string" value="$(arg imu_topic)"/>
    <param name="field_topic" type="string" value="$(arg field_topic)"/>
    <param name="enable_magnetometer" type="bool" value="$(arg enable_magnetometer)"/>
    <param name="broadcast_frame" type="bool" value="$(arg broadcast_frame)"/>

    <rosparam file="$(arg param_file)"/>
  </node>

</launch>
